.github/workflows/format.yml

```diff
@@ -0,0 +1,17 @@
+name: Format check
+
+on: [push, pull_request]
+
+jobs:
+  check:
+    name: Format check
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - uses: astral-sh/ruff-action@v1
+      continue-on-error: true
+      with:
+        args: "format --check"
+        changed-files: "true"
```

Comment:
Nice.

I checked the source code and this relies on https://github.com/tj-actions/changed-files

Years ago, I could not find anything like it _and_ simple and reliable so hopefully this one does the job.

I wrote this `git` script at the time: https://github.com/thesofproject/sof-test/commit/336094b869979bb6fab47158787a6802f4bddad6